week2/homework/src/todoList.js

```diff
@@ -0,0 +1,66 @@
+const fs = require('fs');
+
+const config = require('./config')
+
+const newTodos = new WeakMap();
+const newRead = new WeakMap();
+const newWrite = new WeakMap();
+
+class TodoList {
+  constructor(file) {
+    this.file = file;
+
+    newRead.set(this, file => {
+      const data = fs.readFileSync(file, { encoding: config.ENCODING });
+      return JSON.parse(data);
+    });
+
+    newWrite.set(this, data => {
+      fs.writeFileSync(this.file, JSON.stringify(data, null, 2));
+    });
+    newTodos.set(this, newRead.get(this)(this.file));
+  }
+
+  get todos() {
+    return newTodos.get(this);
+  }
+
+  list() {
+    if (!this.todos.length) throw new Error('No Task');
+    return this.todos;
+  }
+
+  add(task) {
+    if (!task.length) {
+      throw new Error('Task required');
+    }
+    this.todos.push({
+      id: this.todos.length ? Math.max(...this.todos.map(i => i.id)) + 1 : 1,
```

Comment:
👍 good job, you avoided a common mistake to use the length as an id which would result in duplicate ids